projects/plugins/protect/src/class-jetpack-protect.php

```diff
@@ -305,6 +324,37 @@ public function protect_filter_available_modules( $modules ) {
 		return array_merge( array( self::JETPACK_WAF_MODULE_SLUG ), $modules );
 	}
 
+	/**
+	 * Check for user licenses.
+	 *
+	 * @param  boolean $has_license Check if user has a license.
+	 * @param  object  $licenses List of licenses.
+	 * @param string  $plugin_slug The plugin that initiated the flow.
+	 *
+	 * @return boolean
+	 */
+	public static function jetpack_check_user_licenses( $has_license, $licenses, $plugin_slug ) {
+		if ( Plan::has_required_plan() ) {
```

Comment:
```suggestion
		if ( Plan::has_required_plan( true ) ) {
```

I'm having trouble with the first test case, where I:
1. Create a new Jurassic Ninja site running this branch with Jetpack Beta.
2. Connect Jetpack and purchase a Scan plan.
3. Disconnect Jetpack completely via My Jetpack.
4. Return to the Protect interstitial/pricing table and click the new existing plan/license link.
5. Am redirected to the checkout, despite having an attached plan.

I'm guessing (haven't tested) this might be due to `has_required_plan()` caching a `false` value before the plan is purchased.